EIPS/eip-4444.md

```diff
@@ -0,0 +1,101 @@
+---
+eip: 4444
+title: Bound Historical Data in Execution Clients
+description: Prune historical data in clients older than one year
+author: George Kadianakis (@asn_d6), lightclient (@lightclient), Alex Stokes (@ralexstokes)
+discussions-to: TBD
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-11-02
+---
+
+## Abstract
+
+Clients must stop serving historical blocks/receipts that are older than a year on the p2p layer. Clients may prune such historical data.
+
+## Motivation
+
+Historical blocks, states, and receipts currently occupy more than 400GB of disk space (and growing!). Therefore, to validate the chain, users must typically have a 1TB disk.
+
+Historical data is not necessary for validating new blocks, so once a client has synced the tip of the chain, historical data is only retrieved when requested explicitly over the JSON-RPC or when a peer attempts to sync the chain. By pruning the history, this proposal reduces the disk requirements for users. Pruning history also allows clients to remove code that processes historical blocks. This means that execution clients don't need to maintain code paths that deal with each upgrade's compound changes.
+
+This proposal also concretizes the availability guarantees of the p2p layer as clients won't be able to fetch historical data anymore. This results in less strain on the network as clients adopt more lightweight sync strategies based on the PoS weak subjectivity assumption.
+
+## Specification
+
+| Parameter | Value | Description |
+| - | - | - |
+| `HISTORY_PRUNE_EPOCHS` | 82125 | A year in beacon chain epochs |
+
+Clients MUST NOT serve blocks/states/receipts that are older than `HISTORY_PRUNE_EPOCHS` epochs on the p2p network.
+
+Clients MAY locally prune block/state/receipt history that is older than `HISTORY_PRUNE_EPOCHS` epochs.
```

Comment:
Specification should indicate what behavior should occur if a peer asks for data that is too old (which is why I think this should be a Networking spec).

Reply:
Why should anything specific occur? There are a small number of clients, we should work together to ensure that they follow the behavior of the EIP.

Reply:
It's been argued that requesting and serving data beyond the limit should be explicitly incorrect. In such a case, you'd need to have devp2p return errors on such requests